apps/charterafrica/src/lib/openAfrica.js

```diff
@@ -126,14 +127,14 @@ export default async function fetchDatasets(
 
   try {
     const response = await packageSearch(params);
-    const formattedData = formatResponse(response, pathname);
+    const formattedData = formatResponse(response, pathname, locale);
     return formattedData;
   } catch (error) {
     return error;
   }
 }
 
-export async function fetchDataset(id, path) {
+export async function fetchDataset(id, path, locale = "en") {
```

Comment:
```suggestion
export async function fetchDataset(id, pathname, query) {
```

1. `locale` should be part of `query`.
2. use `pathname` consistently ... mixing `path` and `pathname` will lead to confusion down the line.